Counter/src/app/pokemon-service.ts

```diff
@@ -10,9 +10,14 @@ import { ImageData } from './image';
 @Injectable()
 export class PokemonService {
     constructor(private http:HttpClient) { }
-
-    getPokemon() {
-        return POKEMON;
+    
+    getPokemon(): Observable<Pokemon[]> {
+        return of(POKEMON);
+    }
+    getOnePokemon(name: string): Pokemon {
+        let filteredList = POKEMON.find(pokemon => pokemon.name == name);
+        if (filteredList) return filteredList;
+        return new Pokemon;
```

Comment:
lots of ways to reduce the size of this that you can consider. one of which would be
`return filteredList || new Pokemon`